fieldsolver/gridGlue.cpp

```diff
@@ -92,6 +92,118 @@ template <typename T, int stencil> void computeCoupling(dccrg::Dccrg<SpatialCell
+                                 { 6, 12, 18, 12,  6},
+                                 { 3,  6,  9,  6,  3}},
+
+                                 {{ 2,  4,  6,  4,  2},
+                                 { 4,  8, 12,  8,  4},
+                                 { 6, 12, 18, 12,  6},
+                                 { 4,  8, 12,  8,  4},
+                                 { 2,  4,  6,  4,  2}},
+
+                                 {{ 1,  2,  3,  2,  1},
+                                 { 2,  4,  6,  4,  2},
+                                 { 3,  6,  9,  6,  3},
+                                 { 2,  4,  6,  4,  2},
+                                 { 1,  2,  3,  2,  1}}
+                                 };
+
+   // Update momentsGrid Ghost Cells
+   momentsGrid.updateGhostCells(); 
+
+
+   // Get size of local domain and create swapGrid for filtering
+   const int *mntDims= &momentsGrid.getLocalSize()[0];  
+   const int maxRefLevel = mpiGrid.mapping.get_maximum_refinement_level();
+   FsGrid<std::array<Real, fsgrids::moments::N_MOMENTS>, FS_STENCIL_WIDTH> swapGrid = momentsGrid;  //swap array 
+
+   // Filtering Loop
+   for (int blurPass = 0; blurPass < Parameters::maxFilteringPasses; blurPass++){
+
+      // Blurring Pass
+      #pragma omp parallel for collapse(2)
+      for (int k = 0; k < mntDims[2]; k++){
+         for (int j = 0; j < mntDims[1]; j++){
+            for (int i = 0; i < mntDims[0]; i++){
+
+               //  Get refLevel level
+               int refLevel = technicalGrid.get(i, j, k)->refLevel;
+
+               //  Get refLevel level
+               // Skip pass
+               if (blurPass >= P::numPasses[refLevel] ||
+                  technicalGrid.get(i, j, k)->sysBoundaryFlag == sysboundarytype::DO_NOT_COMPUTE ||
+                  (technicalGrid.get(i, j, k)->sysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY &&
+                  technicalGrid.get(i, j, k)->sysBoundaryLayer == 2)
+               )
+               {
+                  continue;
+               }
+
+               // Get pointers to our cells
+               std::array<Real, fsgrids::moments::N_MOMENTS> *cell;  
+               std::array<Real,fsgrids::moments::N_MOMENTS> *swap;
+            
+               // Set Cell to zero before passing filter
+               swap = swapGrid.get(i,j,k);
+               for (int e = 0; e < fsgrids::moments::N_MOMENTS; ++e) {
+                  swap->at(e)=0.0;
+               }
+
+               // Perform the blur
+               for (int c=-kernelOffset; c<=kernelOffset; c++){
+                  for (int b=-kernelOffset; b<=kernelOffset; b++){
+                     for (int a=-kernelOffset; a<=kernelOffset; a++){
+                        cell = momentsGrid.get(i+a,j+b,k+c);
+                        for (int e = 0; e < fsgrids::moments::N_MOMENTS; ++e) {
+                           swap->at(e)+=cell->at(e) *kernel[kernelOffset+a][kernelOffset+b][kernelOffset+c]/kernelSum;
```

Comment:
might be faster to do a division by kernelsum at the end (one division) instead of 27 divisions.

Reply:
Indeed. Ok will handle that.